tests/test_pedigree.py

```diff
@@ -362,6 +362,124 @@ def test_valid_pedigree(self):
+
+    @pytest.mark.parametrize("num_founders", [2, 3, 100, 1000])
+    def test_shallow(self, num_founders):
+        tables = simulate_pedigree(
+            num_founders=num_founders,
+            num_children_prob=[0, 0, 1],
+            num_generations=2,
+        )
+        self.verify(tables)
+
+    @pytest.mark.parametrize("num_founders", [2, 3, 10, 20])
+    def test_deep(self, num_founders):
+        tables = simulate_pedigree(
+            num_founders=num_founders,
+            num_children_prob=[0, 0, 1],
+            num_generations=100,
+        )
+        self.verify(tables)
+
+    @pytest.mark.parametrize("num_founders", [2, 3, 10, 20])
+    def test_many_children(self, num_founders):
+        tables = simulate_pedigree(
+            num_founders=num_founders,
+            num_children_prob=[0] * 99
+            + [1],  # Each pair of parents will always have 100 children
+            num_generations=2,
+        )
+        self.verify(tables)
+
+    @pytest.mark.parametrize("num_founders", [2, 3, 10, 20])
+    def test_unrelated(self, num_founders):
+        tables = simulate_pedigree(
+            num_founders=num_founders,
+            num_children_prob=[0],
+            num_generations=1,
+        )
+        self.verify(tables)
+
+    def test_mixed_generation_parents(self):
+        tables = tskit.TableCollection(sequence_length=1)
+        tables.populations.add_row()
+
+        gen0 = [self.add_individual(tables, time=2) for _ in range(2)]
+        gen1 = [self.add_individual(tables, time=1, parents=gen0)]
+        self.add_individual(tables, time=0, parents=(gen0[0], gen1[0]))
+
+        tables.build_index()
+        self.verify(tables)
+
+    def test_inbreeding_sibs(self):
+        tables = tskit.TableCollection(sequence_length=1)
+        tables.populations.add_row()
+
+        parents = [self.add_individual(tables, time=2) for _ in range(2)]
+        sibs = [self.add_individual(tables, time=1, parents=parents) for _ in range(2)]
+        self.add_individual(tables, time=0, parents=sibs)
+
+        tables.build_index()
+        self.verify(tables)
+
+    def test_half_sibs(self):
+        tables = tskit.TableCollection(sequence_length=1)
+        tables.populations.add_row()
+
+        parents = [self.add_individual(tables, time=1) for _ in range(3)]
+        self.add_individual(tables, time=0, parents=parents[:2])
+        self.add_individual(tables, time=0, parents=parents[1:])
+
+        tables.build_index()
+        self.verify(tables)
+
+    def test_inbreeding_half_sibs(self):
+        tables = tskit.TableCollection(sequence_length=1)
+        tables.populations.add_row()
+
+        parents = [self.add_individual(tables, time=2) for _ in range(3)]
+        half_sib1 = self.add_individual(tables, time=1, parents=parents[:2])
+        half_sib2 = self.add_individual(tables, time=1, parents=parents[1:])
+        self.add_individual(tables, time=0, parents=(half_sib1, half_sib2))
+
+        tables.build_index()
```

Comment:
Is the ``build_index`` needed? Might as well include it in the ``verify`` method if so, rather than doing it n times?